mk/docs.mk

```diff
@@ -288,3 +288,13 @@ doc/style/index.html: $(RUSTBOOK_EXE) $(wildcard $(S)/src/doc/style/*.md) | doc/
 	@$(call E, rustbook: $@)
 	$(Q)rm -rf doc/style
 	$(Q)$(RUSTBOOK) build $(S)src/doc/style doc/style
+
+ERR_IDX_GEN = $(CFG_BUILD)/error-index-generator
+RUSTC_EXE = $(HBIN2_H_$(CFG_BUILD))/rustc
+error-index: doc/error-index.html
+
+doc/error-index.html: src/etc/error-index-generator.rs $(RUSTC_EXE) | doc/
```

Comment:
I think these dependencies aren't quite right because this needs to depend on libraries like libsyntax and librustdoc from stage2. I think you'll want to depend on `CSREQ2_T_$(CFG_BUILD)_H_$(CFG_BUILD)` here.